tests/rptest/services/redpanda.py

```diff
@@ -646,3 +646,38 @@ def delete_topic(self, name):
 
     def cov_enabled(self):
         return self._context.globals.get(self.COV_KEY, self.DEFAULT_COV_OPT)
+
+    def transfer_partition_leadership(self,
+                                      topic,
+                                      partition,
+                                      target,
+                                      timeout_sec=30):
+        kc = KafkaCat(self)
+
+        # source / target nodes
+        target = self.idx(target) if isinstance(target,
+                                                ClusterNode) else target
+        source, _ = kc.get_partition_leader(topic, partition, timeout_sec=30)
+        self.logger.debug(
+            f"Transferring leadership source {source} target {target}")
+
+        # build admin url
+        source_node = self.get_node(source)
+        url = "http://{}:9644/v1/partitions/kafka/{}/{}/transfer_leadership?target={}".format(
+            source_node.account.hostname, topic, partition, target)
+
+        def try_transfer():
+            self.logger.debug(url)
+            res = requests.post(url)
```

Comment:
This is the same logic as in the original place, but it was wrong (this code shouldn't be issuing multiple POSTs if the transfer doesn't complete promptly).  Cleaned up in the redirect PR here (https://github.com/vectorizedio/redpanda/pull/2806/commits/99575f8e9922eef4d047f0955077eb95c822f62f)

(also the `res` is never being checked for success).

Kind of orthogonal to this PR, but feels a bit dangerous to pull the bad code into a shared place it's likely to get reused, would be good to fix it at same time.

Reply:
ah forgot about the requests stuff. thanks will deal with this

Reply:
@jcsp looking at this again it seems like I should wait for the redirect PR to merge and then reuse the fixed transfer logic you added?

Reply:
@dotnwat yep.  Good news is it's merged now!

Reply:
fixed